### PR TITLE
Move helper methods from ReTheme to jsutils

### PR DESCRIPTION
**Pull request by lancetipton**
## Context

* There were a few methods that should go into jsutils instead of ReTheme
* We should move them to jsutils

## Goal

* Move some methods form ReTheme to  jsutils 

## Updates
* Added a couple array helper methods
  * `flatArr` - Makes an array with arrays flat 
  * `ensureArr` - Ensures a variable is an array
* Migrated some methods from ReTheme to jsutils
  *  `hashstring` - Converts a string in to a hash
  * `hyphenator` - Converts a CSS in JS rule  string to standard CSS style with hyphens  
* Updated calls to `hasDomAccess` to use the jsutils version

## Testing

**Test 1**
* Pull the PR => `keg pr 115`
* Run tests for jsutils => `keg jsutils && yarn test`
* Run tests for ReTheme => `keg retheme && yarn test`

**Test 2**
* Run this `Re-theme` docker package => `keg retheme pack run docker.pkg.github.com/simpleviewinc/keg-packages/retheme:add-storybook-to-retheme`
  * In your browser navigate to http://retheme-add-storybook-to-retheme.local.kegdev.xyz
  * Ensure everything works as expected

**Test 3**
* Run this `Keg-Components` docker package => `keg dpg run docker.pkg.github.com/simpleviewinc/keg-packages/keg-components:add-storybook-to-retheme`
  * This has the `jsutils` and  `ReTheme` changes synced into it
*  In your browser navigate to http://components-add-storybook-to-retheme.local.kegdev.xyz/
  * Ensure **ALL** components render as expected

### CODE REVIEW

**Review by mpcarolin — CHANGES_REQUESTED** [review]
ReTheme's image passed testing. When I ran the `keg-components` one though, I encountered this for several of the components:

image

Doesn't really seem related to this at all, so not sure what's happening there.

I also left one code suggestion regarding the hash function.

File: repos/jsutils/src/string/hashString.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * <br/> Value comes from being a pure function
+ * <br/> Given the same input, it will always return the same output
+ * <br/> There is no expectation to convert back from the hash to the original string
+ * @function
+ * @param {string} str - String to be hashed
+ * @param {number=} maxLength - Max length of the returned hash
+ *
+ * @returns {string} - Hashed version of the string
+ */
+export const hashString = (str, maxLength=0) => {
```
[the remark this answers is not in the document]
Comment (lancetipton) [reply repos/jsutils/src/string/hashString.js:17]: Yah, that works, I'll update to that

[the remark this answers is not in the document]
Comment (lancetipton) [reply repos/jsutils/src/string/hashString.js:17]: Actually looking at it, why would I ever want a 0 length string? 
I think it's correct the way it is. Makes no sense to even call it if maxLength is set to 0

**Review by mpcarolin — APPROVED** [review]
Looks good 👍🏼

**Review by daniel112 — APPROVED** [review]
all unit test passed and the storybook looks good

**Pull request merged**